test/helpers/contract-helpers.ts

```diff
@@ -164,89 +165,165 @@ export const initializeContract = async () => {
+  );
 
   return withdrawnCluster.eventsByName.ClusterWithdrawn[0].args;
 };
 
 export const reactivate = async (ownerId: number, operatorIds: number[], amount: string, cluster: any) => {
   await DB.ssvToken.connect(DB.owners[ownerId]).approve(DB.ssvNetwork.contract.address, amount);
-  const reactivatedCluster = await trackGas(DB.ssvNetwork.contract.connect(DB.owners[ownerId]).reactivate(
-    operatorIds,
-    amount,
-    cluster));
+  const reactivatedCluster = await trackGas(
+    DB.ssvNetwork.contract.connect(DB.owners[ownerId]).reactivate(operatorIds, amount, cluster),
+  );
   return reactivatedCluster.eventsByName.ClusterReactivated[0].args;
 };
 
-export const registerValidators = async (ownerId: number, numberOfValidators: number, amount: string, operatorIds: number[], gasGroups?: GasGroup[]) => {
+export const registerValidatorsWithKeystores = async (
+  ownerId: number,
+  amount: string,
+  keystores: string[],
+  passwords: string[],
+  operatorIds: number[],
+  operatorKeys: string[],
+  gasGroups?: GasGroup[],
+) => {
+  const validators: any = [];
+  let args: any;
+
+  if (keystores.length === passwords.length && operatorIds.length === operatorKeys.length) {
+    // keystores.length is same with the number of validators, because each validator has a corresponding keystore file.
+
+    const operators = operatorKeys.map((operatorKey, index) => ({
+      id: operatorIds[index],
+      operatorKey,
+    }));
+
+    const ssvKeys = new SSVKeys();
+
+    // Register validators to contract
+    for (let i = 0; i < keystores.length; i++) {
+      const { publicKey, privateKey } = await ssvKeys.extractKeys(keystores[i], passwords[i]);
+
+      const threshold = await ssvKeys.createThreshold(privateKey, operators);
+      const encryptedShares: EncryptShare[] = await ssvKeys.encryptShares(operators, threshold.shares);
+
+      const keyShares = new KeyShares();
+      const payload = await keyShares.buildPayload(
+        {
+          publicKey,
+          operators,
+          encryptedShares,
+        },
+        {
+          ownerAddress: DB.owners[ownerId].address,
+          ownerNonce: 1,
+          privateKey,
+        },
+      );
+
+      await DB.ssvToken.connect(DB.owners[ownerId]).approve(DB.ssvNetwork.contract.address, amount);
+      const result = await trackGas(
+        DB.ssvNetwork.contract
+          .connect(DB.owners[ownerId])
+          .registerValidator(payload.publicKey, payload.operatorIds, payload.sharesData, amount, {
+            validatorCount: 0,
+            networkFeeIndex: 0,
+            index: 0,
+            balance: 0,
+            active: true,
+          }),
+        gasGroups,
+      );
+      args = result.eventsByName.ValidatorAdded[0].args;
+      DB.validators.push(payload);
+      validators.push({ publicKey: payload.publicKey, shares: payload.sharesData });
+    }
+  }
+
+  return { validators, args };
+};
+
+export const registerValidators = async (
```

Comment:
I think this can be refactored to use `registerValidatorsWithKeystores`. For each `numberOfValidators` pick it from the validatorsData JSON file and call `registerValidatorsWithKeystores`...